test/integration/contract.js

```diff
@@ -319,33 +320,34 @@ describe('Contract', function () {
     result.callerId.should.be.equal(DRY_RUN_ACCOUNT.pub)
   })
 
-  it('calls deployed contracts', async () => {
-    const result = await deployed.call('getArg', ['42'])
-    return result.decode().should.eventually.become(42)
+  it('calls deployed contracts with unsafe integer', async () => {
+    const unsafeInt = BigInt(Number.MAX_SAFE_INTEGER + '0')
+    const result = await deployed.call('getArg', [unsafeInt])
+    expect(result.decodedResult).to.be.equal(unsafeInt)
   })
 
   it('call contract/deploy with waitMined: false', async () => {
     const deployed = await bytecode.deploy([], { waitMined: false })
     await sdk.poll(deployed.transaction, { interval: 50, attempts: 1200 })
     expect(deployed.result).to.be.equal(undefined)
     deployed.txData.should.not.be.equal(undefined)
-    const result = await deployed.call('getArg', ['42'], { waitMined: false })
+    const result = await deployed.call('getArg', [42], { waitMined: false })
     expect(result.result).to.be.equal(undefined)
     result.txData.should.not.be.equal(undefined)
     await sdk.poll(result.hash, { interval: 50, attempts: 1200 })
   })
 
   it('calls deployed contracts static', async () => {
-    const result = await deployed.callStatic('getArg', ['42'])
-    return result.decode().should.eventually.become(42)
+    const result = await deployed.callStatic('getArg', [42])
+    expect(result.decodedResult).to.be.equal(42n)
```

Comment:
why is this with 'n' here?

Reply:
it's `BigInt` which is used in the calldata lib right now. this is also sth. that should be addressed with the integration. I have some open issues in that regards, see:
- https://github.com/aeternity/aepp-sdk-js/issues/1290

Reply:
we should figure this out or provide a different solution with chai if this is the issue

Reply:
@davidyuk can you verify the problem and comment this again?

Reply:
I've created a new project with the latest mocha (9.1.3) and chai (4.3.4):
```js
import { expect } from 'chai'

it('works', () => {
    expect({ a: 1n }).to.be.eql({ a: 2n })
})
```
`mocha test.js` outputs:
```
  1) works:

      AssertionError: expected { a: 1n } to deeply equal { a: 2n }
      + expected - actual

       {
      -  "a": "1"
      +  "a": "2"
       }
```
in the diff section at the end BigInts presented as strings for some reason. If this is the issue, I don't think that we should do anything on our side to avoid that.

Reply:
```js
expect({ a: '1' }).to.be.eql({ a: 2n })
```
```
       {
      -  "a": "1"
      +  "a": "2"
       }
```
🤷‍♀️

Reply:
I still don't get why we have to have it like this. If it stays like this there needs to be a big explainer everywhere for ALL aepp devs to adjust their aepps

Reply:
you mean the `BigInt` in general? well I think in the past we just weren't able to use it and in the calldata lib it makes definitely sense to use it as it is possible now.

I can understand your point of view. but isn't `BigInt` the right way to do it?

@thepiwo wouldn't you have to touch all repos again anyway with the upcoming AEproject release or how will that transition look like again?  (can't exactly remember)

I mean it's definitely a bit unfortunate but it will be a major release - so we could (and maybe should) do it this way 🤷‍♀️

Reply:
No, repos that don't use aeproject don't have to be touched, e.g. governance-server, governance-aepp, superhero wallet, graffiti-aepp, graffiti-server.

We should just make the use of BigInt very clear in docs and Changelogs and so on, I'm not saying its bad to have it.